### ISSUE TITLE
`someUTxO` does not actually update `envUsedSomeUTxOs` in `GYTxMonadNode`

### ISSUE THREAD

**Issue by TotallyNotChase** [geniusyield/atlas#318]
**Describe the bug**
If we look at the current `someUTxO` implementation for `GYTxMonadNode`:
```hs
someUTxO lang = do
    addrs           <- ownAddresses
    utxosToConsider <- availableUTxOs
    case lang of
      PlutusV2 ->
        case someTxOutRef utxosToConsider  of
            Just (oref, _) -> return oref
            Nothing        -> throwError . GYQueryUTxOException $ GYNoUtxosAtAddress addrs
      PlutusV1 ->
        case find utxoTranslatableToV1 $ utxosToList utxosToConsider of
          Just u  -> return $ utxoRef u
          Nothing -> throwError . GYQueryUTxOException $ GYNoUtxosAtAddress addrs  -- TODO: Better error message here?
```
We notice that the returned UTxO isn't actually removed from `envUsedSomeUTxO`. This seems problematic since the point of `envUsedSomeUTxO` is to track previously yielded `someUTxO`s so they aren't returned repeatedly.

As it currently stands, the field `envUsedSomeUTxO` is actually useless.

**Expected behavior**

`envUsedSomeUTxO` should be updated accordingly.

**Additional context**

Has this always been like this? I thought it used to work.

**Comment by TotallyNotChase** [comment]
Tangentially related but the framework should use `availableUTxOs` wherever possible to obtain own utxos, instead of querying the own addresses.

**Comment by TotallyNotChase** [comment]
Also, `someUTxO` impl for `GYTxMonadNode` should really be replaced with the `randomTxOutRef` from `GYUTxO`

**Comment by sourabhxyz** [comment]
> As it currently stands, the field envUsedSomeUTxO is actually useless.

That is true, I think it never served any meaningful purpose and was just there in case we wanted to give it some meaning.

> We notice that the returned UTxO isn't actually removed from envUsedSomeUTxO. This seems problematic since the point of envUsedSomeUTxO is to track previously yielded someUTxOs so they aren't returned repeatedly.

I think you meant that it is not _added_ to `envUsedSomeUTxOs`?
